### PR TITLE
[READY FOR DESIGN REVIEW] -> MachineLearningServices: Adding new API version and definition for AmlInstance compute type

### PR DESCRIPTION
**Pull request by vhamine**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by Juliehzl** [comment]
Hi @vhamine, why do you send a PR in draft? Are you ready for review process?

**Comment by vhamine** [comment]
> Hi @vhamine, why do you send a PR in draft? Are you ready for review process?

**Comment by vhamine** [comment]
> Hi @vhamine, why do you send a PR in draft? Are you ready for review process?

The PR is not ready for review yet as I am still making changes. This being my first interaction with azure rest API specification, I thought I should create a draft PR and get familiar with the pipeline checks. I plan to attend the office hours today and send a PR for review by EOD. Thanks!

**Comment by vhamine** [comment]
This PR achieves the following:
- It defines a new API version 2019-11-01.

- It adds a new ComputeType to MachineLearningServices Workspace RP. The new compute type is called "AmlInstance".

- It defines input specification for the new compute type. This follows the same model as existing compute types as well as same request/response structure.

- It adds Windows OS support for existing AmlCompute compute type.

**Comment by Juliehzl** [comment]
> * It adds a new ComputeType to MachineLearningServices Workspace RP. The new compute type is called "AmlInstance".
> *

Thanks for your detailed information. In this way, we can start review process now.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json
Comment (oolukoya) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: We should perhaps add to the description that the default time is 2am. Let me double check on this.

Comment (cyly) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: Could you change it to datastoresMountSettings? In datastore service, they try to use datastore instead of DataStore.

Comment (cyly) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: The same as above. Please apply to all the other changes. Thank you!

Comment (cyly) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: Let's remove the could be a URI. We only need to save the relative path so that later it is easier to do migration.

Comment (cyly) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: If it is maximum is 30 days, why int64? Could we use int32 here?

Comment (cyly) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: The same. If the maximum is 0-23, why int64?

Comment (cyly) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: I think Femi wants to add a version here. Please confirm with Femi.

**Review by ghost — CHANGES_REQUESTED** [review]

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json
Comment (cyly) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: I talked with Ahmet and Ahmet suggests not adding this. If needed, we can add it later and it will not be a breaking change. Please remove this property.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json
Comment (cyly) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: Do you need to follow the convention to define publickey in the properties. Also as what Chris said, you need to add a readonly property for the admin name. Otherwise there is no way to tell users the admin user name.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/examples/createComputeInstance.json
Comment (cyly) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/examples/createComputeInstance.json]: I didn't see this defined in the amlInstance request and response. Do you miss that in the definition?

Comment (vhamine) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/examples/createComputeInstance.json]: Yu, as discussed, you were not looking at the latest version.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json
Comment (sanjaiganesh) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: >"osType": { [](start = 16, length = 11)

add default 

 "default": Linux

Comment (sanjaiganesh) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: >applicationUris [](start = 17, length = 15)

rename.. applicationUris => applications

Comment (sanjaiganesh) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: >ComputeInstanceApplicationUri [](start = 5, length = 29)

rename : ComputeInstanceApplicationUri => ComputeInstanceApplication

Comment (sanjaiganesh) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: strip off application from applicationName and applicationUri  

and consider adding icon for display purposes.. having this in mind, i would suggest

"displayName"
"endpointUri"
"displayIcon" [future]

Comment (vhamine) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: Thanks, done!

**Comment by vhamine** [comment]
@Juliehzl this PR was signed-off by ARM. We do not want to merge this PR yet since our back-end is still being implemented. We should co-ordinate merging this before Ignite 2019. Can you please let us know what the next steps should be?

CC: @ahcan76

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json
Comment (Juliehzl) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json]: Where is your nextLink property for nextLinkName?

**Review by Juliehzl — APPROVED** [review]

**Comment by Juliehzl** [comment]
Hi @vhamine, I have a look at your swagger specification. It looks good to me and I already approved for your PR. All review process is completed before clicking the merge button.
If you don't want it to be merged now, I will add DONOTMERGE label for your PR and when you are ready to merge, just let me know and I will click the merge button for you.

**Comment by akning-ms** [comment]
Hi @vhamine, Can we know what is this PR status as it is existing for a long time？

**Comment by ramagott** [comment]
> Hi @vhamine, Can we know what is this PR status as it is existing for a long time？

This PR is not required any more. Can you close it?

**Pull request closed without merging**
